src/tools/ScenarioMeasurement/Startup.Tests/StartupTests.cs

```diff
@@ -0,0 +1,99 @@
+using Microsoft.DotNet.PlatformAbstractions;
+using Reporting;
+using ScenarioMeasurement;
+using System;
+using System.CommandLine.Invocation;
+using System.Diagnostics;
+using System.IO;
+using System.Resources;
+using Xunit;
+
+namespace Startup.Tests
+{
+    public class StartupTests
+    {
+        Logger logger = new Logger("test-startup.log");
+        string traceDirectory = Environment.CurrentDirectory;
+
+        [WindowsOnly]
+        public void TestWindowsTraceSession()
+        {
+            string sessionName = "test-windows-session";
+            string traceName = "test-windows-trace";
+            var session = new WindowsTraceSession(sessionName, traceName, traceDirectory, logger);
+            var parser = new TimeToMainParser();
+            TestSession(session, parser);
+        }
+
+        [LinuxOnly]
+        public void TestLinuxTraceSession()
+        {
+            string sessionName = "test-linux-session";
+            string traceName = "test-linux-trace";
+            var session = new LinuxTraceSession(sessionName, traceName, traceDirectory, logger);
+            var parser = new TimeToMainParser();
+            TestSession(session, parser);
+        }
+
+        [Fact]
+        public void TestProfileIteration()
+        {
+            string sessionName = "test-profile-iteration-session";
+            string traceName = "test-profile-iteration-trace";
+            var timeToMainParser = new TimeToMainParser();
+            var profileParser = new ProfileParser(timeToMainParser);
+            var profileSession = TraceSessionManager.CreateProfileSession(sessionName, traceName, traceDirectory, logger);
+            TestSession(profileSession, profileParser);
+        }
+
+        private void TestSession(ITraceSession session, IParser parser)
+        {
+            ProcessHelper.Result result;
+            string traceFilePath = "";
+            using (session)
+            {
+                session.EnableProviders(parser);
+                result = RunTestIteration();
+                traceFilePath = session.GetTraceFilePath();
+            }
+            if (result != ProcessHelper.Result.Success)
+            {
+                throw new Exception("Test iteration failed.");
+            }
+            Assert.True(!String.IsNullOrEmpty(traceFilePath) && File.Exists(traceFilePath));
```

Comment:
You can have more than one assertion. In this formulation you don't know which failed. I would use an `Assert.False` on the first condition and make the second its own `Assert.True`

Reply:
Not blaming it but I saw this use of combined logic in TraceEvent library so I copied that. Learned something bad.

Reply:
This is specific advice for tests. Having assertions be very granular makes the diagnosis of failures easier because the failure can only mean one thing. In normal code this would have been just fine.